Add junit xml support

cddr opened the pull request:
This PR refactors the `runner-fn` a little to give it access to the parsed cli arguments. That makes it possible to make the junit option behave more like the other report options in that you just have to enable it and it will write to the same output directory as all the other reports.

I think this is basically what @palfrey meant with his comment in #80 

This would resolve #116

rfkm commented on cloverage/src/cloverage/coverage.clj:
```diff
@@ -98,6 +99,8 @@
        ["--[no-]coveralls"
         "Send a JSON report to Coveralls if on a CI server" :default false]
+       ["--[no-]junit"
+        "Output test results as junit xml file" :default false]
```
You might want to document this option works only for the `clojure.test` runner.
It would be nicer if users can see warnings or errors by specifying an incompatible combination of runners and reporters. While we can introduce such a generic mechanism, I think ad-hoc handling is enough for now.

cddr commented:
Sounds good. I haven't given much thought to the different capabilities of test runners that cloverage would be interested in. I updated the PR to throw an error when junit is combined with non-clojure.test  runner. I also cleaned up my own commits a little bit but kept @ryfow's commit in-tact. Thanks for the original work.

cddr commented:
Looking forward to being able to use this in our circle builds. Anything blocking it at this point?

lvh commented:
Nope, just maintainer time :)

cddr commented:
Thanks! :-)
